inference-engine/thirdparty/clDNN/kernel_selector/core/common/jitter.cpp

```diff
@@ -738,6 +738,17 @@ JitConstants MakeActivationJitConstants(ActivationFunction activation_function,
                     (input / (one + exp(neg(input)))).str()));
             break;
         }
+        case ActivationFunction::MISH: {
+            std::string type_suffix = out_dt == Datatype::F32 ? "f" : "h";
+            const JitTerm twenty("20." + type_suffix);
```

Comment:
@Lyamin-Roman I suppose exp threshold should be smaller for fp16. Max representable fp16 value is ~65504 (depends on rounding mode). So threshold for single exponent must be ~11 to get meaningful value. But since this approximation requires calculation of `x * exp(x) * (exp(x) + 2)` value, the threshold should be like 4.2-4.3. 

Also I'm a bit concerned that it's pretty easy to get inf output for this approximation, so maybe it makes sense to add one more clamp for FP16 at least like `min(input * exp(x) * (exp(x) + 2), HALF_MAX) / (exp(x) * (exp(x) + 2) + 2)` @dkurt  what do you think about it?